backend/tests/v5/e2e/routes/sso/aad/aad.test.js

```diff
@@ -71,8 +76,47 @@ const testAuthenticate = () => {
 
 const testAuthenticatePost = () => {
 	describe('Authenticate Post', () => {
+		test('should fail if state is not provided', async () => {
+			const res = await agent.get('/v5/sso/aad/authenticate-post')
+				.expect(templates.unknown.status);
+			expect(res.body.code).toEqual(templates.unknown.code);
+		});
+
+		test('should redirect and add error to the query if user does not exist', async () => {
+			const userDataFromAad = { mail: generateRandomString(), id: generateRandomString() };
+			const state = { redirectUri: generateRandomURL() };
+			Aad.getUserDetails.mockResolvedValueOnce({ data: userDataFromAad });
+			const res = await agent.get(`/v5/sso/aad/authenticate-post?state=${encodeURIComponent(JSON.stringify(state))}`)
+				.expect(302);
+			expect(res.headers.location).toEqual(`${state.redirectUri}?error=${errorCodes.userNotFound}`);
```

Comment:
Can we add a test to ensure if there's query string on the provided redirectUri we're retaining it?

i..e if the redirectUri is `https://www.abc.com/?blah=blah` then these error codes should still append correctly